File: psup assignment/java/q17.java
import java.util.*;
class Calculator {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter operator (+, -, *, /): ");
        char operator = input.next().charAt(0);
        System.out.print("Enter first number: ");
        double num1 = input.nextDouble();
        System.out.print("Enter second number: ");
        double num2 = input.nextDouble();
        switch (operator) {
            case '+': System.out.println("Result: " + (num1 + num2)); break;
            case '-': System.out.println("Result: " + (num1 - num2)); break;
            case '*': System.out.println("Result: " + (num1 * num2)); break;
            case '/': System.out.println("Result: " + (num1 / num2)); break;
            default: System.out.println("Invalid operator"); break;
        }
        input.close();
    }
}

File: psup assignment/cpp/q17.cpp
#include <iostream>
#include <string>
#include <cctype>
int main() {
    char op;
    double num1, num2;
    std::cout << "\n--- Program 17: Calculator ---\n";
    std::cout << "Enter operator (+, -, *, /): ";
    std::cin >> op;
    std::cout << "Enter first number: ";
    std::cin >> num1;
    std::cout << "Enter second number: ";
    std::cin >> num2;
    switch (op) {
        case '+': std::cout << "Result: " << num1 + num2 << std::endl; break;
        case '-': std::cout << "Result: " << num1 - num2 << std::endl; break;
        case '*': std::cout << "Result: " << num1 * num2 << std::endl; break;
        case '/': std::cout << "Result: " << num1 / num2 << std::endl; break;
        default: std::cout << "Invalid operator" << std::endl; break;
    }
    return 0;
}